### ISSUE TITLE
ThreadPool segfault

### ISSUE THREAD

**Issue by emanuele-f** [ntop/ntopng#1526]
```
Thread 8 "ntopng" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe7269700 (LWP 9945)]
0x000000000041b6be in ThreadPool::dequeueJob (this=0x8a3820, waitIfEmpty=true) at src/ThreadPool.cpp:130
130	  t = threads.front();
(gdb) backtrace 
#0  0x000000000041b6be in ThreadPool::dequeueJob (this=0x8a3820, waitIfEmpty=true)
    at src/ThreadPool.cpp:130
#1  0x000000000041b570 in ThreadPool::run (this=0x8a3820) at src/ThreadPool.cpp:82
#2  0x000000000041b268 in doRun (ptr=0x8a3820) at src/ThreadPool.cpp:29
#3  0x00007ffff69cf6ba in start_thread (arg=0x7fffe7269700) at pthread_create.c:333
#4  0x00007ffff46c73dd in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by ryny24** [comment]
Is this the same issue as mine? I upgraded today and it won't run:

probe@cust225-240-78-64:~$ /usr/local/bin/ntopng /etc/ntopng/ntopng.conf >> /var/log/ntopng/ntopng.log
Segmentation fault

I'm trying to downgrade but haven't been able to yet.

**Comment by emanuele-f** [comment]
Hi, if the segfault happens at startup, then it's probably fixed with ccb4c2cf1947b78dc4ecba91b9fb15a3b8d884cb . This is a different bug

**Comment by ryny24** [comment]
Ok I am running again. I was able to downgrade.
sudo dpkg -i ntopng_3.1.171018-3577_amd64.deb
dpkg: warning: downgrading ntopng from 3.1.171019-3590 to 3.1.171018-3577

**Comment by emanuele-f** [comment]
Hi, a new package containing the fix is now available, please update to new version.

This bug should be fixed with commit 2e8be3ee97ce141903805363c811b7096415af0b

**Comment by simonemainardi** [comment]
I confirm the issue is fixed as of the latest dev build.
